exercises/luhn/canonical-data.json

```diff
@@ -29,6 +34,36 @@
       "description": "valid strings with a non-digit added become invalid",
       "input": "046a 454 286",
       "expected": false
+    },
+    {
+      "description": "punctuation is not allowed",
+      "input": "055-444-285",
+      "expected": false
+    },
+    {
+      "description": "symbols are not allowed",
+      "input": "055£ 444$ 285",
+      "expected": false
+    },
+    {
+      "description": "single zero with space is invalid",
+      "input": " 0",
+      "expected": false
+    },
+    {
+      "description": "lots of zeros are valid",
+      "input": " 00000",
+      "expected": true
+    },
+    {
+      "description": "another valid sin",
+      "input": "055 444 285",
+      "expected": true
+    },
+    {
+      "description": "nine doubled is nine",
```

Comment:
in case anyone wonder, this caught a bug in mine (I was using mod 9 after doubling, which turns nine into zero instead of nine)